apps/server/src/modules/task/uc/task-copy.uc.ts

```diff
@@ -26,12 +26,24 @@ export class TaskCopyUC {
 			throw new NotFoundException('could not find task to copy');
 		}
 
-		const destinationCourse = await this.getDestinationCourse(parentParams.courseId, user);
+		const destinationCourse = await this.getDestinationCourse(parentParams.courseId);
+		if (parentParams.courseId) {
+			await this.authorisation.checkPermissionByReferences(
+				userId,
+				AllowedAuthorizationEntityType.Course,
+				parentParams.courseId,
+				{
+					action: Actions.write,
+					requiredPermissions: [],
```

Comment:
maybe we should check some permissions?

Reply:
This check was done before in https://github.com/hpi-schul-cloud/schulcloud-server/pull/3897/files/34c54a203c90ad59627564681ecda53c3d7bcf21#diff-02c32fa3614ad128f280bf08e03fa518c3244d34fe7b3a2dbb8ab2905382635dL59-L61 and it didn't have any permissions (the same is already on lesson.copy.uc.ts). I changed this check because in the future you will need to remove courseRepo from here anyway (this requires changes to the entire project structure, modules, etc. so it won't be done in this PR)